core/state_update.go

```diff
@@ -51,38 +52,91 @@ func (d *StateDiff) Length() uint64 {
 }
 
 func (d *StateDiff) Hash() *felt.Felt {
```

Comment:
even though I called it `Hash()` essentially it's a state diff commitment right? Is it ok that right now we have `Commitment` method for old hash calculation and now it's called `Hash` ?

Reply:
Right now there 2 different way to create commitment for statediff, one is done by the feeder when serving get_signature request while the starknet-api does the other for block building. For now we have to keep both